File: backend/app/schemas/community.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.schemas.user import UserPublic


class ServiceCreate(BaseModel):
    category: str
    title: str
    description: str
    is_paid: bool = False
    price_info: str | None = None
    mode: str = "remote"
    location: str | None = None


class ServiceOut(BaseModel):
    id: uuid.UUID
    category: str
    title: str
    description: str
    is_paid: bool
    price_info: str | None
    mode: str
    location: str | None
    is_active: bool
    created_at: datetime
    provider: UserPublic

    model_config = {"from_attributes": True}


class ProgramCreate(BaseModel):
    category: str
    title: str
    description: str
    event_date: datetime
    is_online: bool = True
    location: str | None = None
    capacity: int | None = None
    is_free: bool = True
    price_info: str | None = None


class ProgramOut(BaseModel):
    id: uuid.UUID
    category: str
    title: str
    description: str
    event_date: datetime
    is_online: bool
    location: str | None
    capacity: int | None
    is_free: bool
    price_info: str | None
    created_at: datetime
    organizer: UserPublic
    rsvp_count: int
    rsvped_by_me: bool

    model_config = {"from_attributes": True}

File: frontend/src/types/community.ts
import type { User } from "./user";

export type ServiceCategory =
  | "visa" | "legal" | "finance" | "tax" | "career"
  | "teaching" | "language" | "housing" | "tech" | "other";

export type ServiceMode = "remote" | "in_person" | "both";

export interface Service {
  id: string;
  category: ServiceCategory;
  title: string;
  description: string;
  is_paid: boolean;
  price_info: string | null;
  mode: ServiceMode;
  location: string | null;
  is_active: boolean;
  created_at: string;
  provider: User;
}

export type ProgramCategory =
  | "workshop" | "meetup" | "study_group" | "networking"
  | "cultural" | "language" | "webinar" | "other";

export interface Program {
  id: string;
  category: ProgramCategory;
  title: string;
  description: string;
  event_date: string;
  is_online: boolean;
  location: string | null;
  capacity: number | null;
  is_free: boolean;
  price_info: string | null;
  created_at: string;
  organizer: User;
  rsvp_count: number;
  rsvped_by_me: boolean;
}
